_posts/2021-04-25-ocp-java11.md

```diff
@@ -0,0 +1,290 @@
+
+For the exam you'll learn all the command line switches and their inconsistencies when they use single or double hyphen.
+Also options for commands like `jmod`.
+Actually I did not know that command before.
+So my taking was here that this command exists and roughly its purpose.
+But why you need to memorize which command line options it offers and get tricked by options that might look right?
+When I need to know what option a command offers, I'll ask for it in a terminal.
+Of course you should be aware of the most important commands and the options you would most likely use.
+It does not make you a better programmer to know options you very seldomly would use.
+
+I learned a lot on Java serialization.
+Actually more then I would like to.
+Nowadays I don't really care much about Java object serialization.
+The basic rules are well understood.
+It was quite interesting to learn the order of object initialization, i.e. which constructor is called or not and in which order.
+I never really cared so far.
+I read about options to influence what and how state is (de-) serialized which I'll never use again.
+However, now I know there are more options available than I was aware of before.
+
+Of course there were also topics which I use daily, but where the book really helped me to learn and think about in detail. I especially enjoyed here
+the chapters about Generics, Functional Programming, Modular Applications, Concurrency and NIO.2. This really helped me to improve my hard skills.
+
+## Mock Exams
+
+Additionally to the Study Guide I ordered the [1Z0-819-J Mock Exams from Enthuware](https://enthuware.com/java-certification-mock-exams/oracle-certified-professional/ocp-java-11-exam-1Z0-819-J).
+I read about them when I was looking into some threads about the certification in the [OCP Forum at Coderanch](https://coderanch.com/f/24/java-programmer-OCPJP).
+I was at this forum to report some small possible errata in the book.
+In the threads some developers were mentioning the mock exams and I thought it could be worth to try some additionally to the review questions in the book.
+
+Before the exam I just had to do 7 of those mock exams (there are 22 with ~50 questions each).
+After doing this I can really recommend to use them when preparing for the exam.
+While it is possible here to get offered how many answers you actually have to choose, the questions themselves felt partially even trickier than in the book.
+
+You can either review them after finishing the complete test or choose to be able to review after each question.
+I started with the first option and reviewed the questions where I failed.
+The different options are commented and help to understand the topic in more detail.
+For the last mock exams I decided to use the option to review after each question, and used this to understand also for correct answers if they were answered correctly because I really hit the nail there.
```

Comment:
```suggestion
For the last mock exams I decided to use the option to review after every question, and used this to understand also for correct answers if they were answered correctly because I really understood the topic.
```